File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task58_PERPLEXITY_llama-3-sonar-large-32k-chat.java
public class Main {
    public static boolean xo(String str) {
        int xCount = 0;
        int oCount = 0;
        for (char c : str.toLowerCase().toCharArray()) {
            if (c == 'x') xCount++;
            if (c == 'o') oCount++;
        }
        return xCount == oCount;
    }

    public static void main(String[] args) {
        System.out.println(xo("ooxx")); // true
        System.out.println(xo("xooxx")); // false
        System.out.println(xo("ooxXm")); // true
        System.out.println(xo("zpzpzpp")); // true
        System.out.println(xo("zzoo")); // false
    }
}

File: dataset/output/PERPLEXITY_llama-3-sonar-large-32k-chat/Task58_PERPLEXITY_llama-3-sonar-large-32k-chat.c
#include <stdio.h>
#include <string.h>
#include <ctype.h>

int xo(char *str) {
    int xCount = 0;
    int oCount = 0;
    for (int i = 0; i < strlen(str); i++) {
        if (tolower(str[i]) == 'x') xCount++;
        if (tolower(str[i]) == 'o') oCount++;
    }
    return xCount == oCount;
}

int main() {
    printf("%d
", xo("ooxx")); // 1
    printf("%d
", xo("xooxx")); // 0
    printf("%d
", xo("ooxXm")); // 1
    printf("%d
", xo("zpzpzpp")); // 1
    printf("%d
", xo("zzoo")); // 0
    return 0;
}